src/main/java/com/allocadia/carbonite/PersistenceInfo.java

```diff
@@ -1,47 +1,110 @@
+    private final String tableName;
     private final String idField;
-    private final ImmutableMap<String, Field> column2field;
+    private final Map<String, Field> column2field;
+    private final Map<Field, String> field2Column;
+    private final Map<String, Field> fieldName2Field;
 
-    public PersistenceInfo(Class<T> clazz, ImmutableMap<String, Field> fields, String idField) {
-        this(fields, Objects.requireNonNull(idField), Objects.requireNonNull(clazz));
+    public static <T> PersistenceInfo<T> from(Class<T> clazz) {
+        return new PersistenceInfo<>(clazz, getTableName(clazz), getFields(clazz), getIdField(clazz));
+    }
+    
+    private PersistenceInfo(Class<T> clazz, String tableName, ImmutableBiMap<String, Field> fields, String idField) {
+        this(fields, Objects.requireNonNull(idField), Objects.requireNonNull(clazz), tableName);
 
         fields.values().forEach(field -> field.setAccessible(true));
     }
 
     //Private version that does not do any validation / modifying of fields
-    private PersistenceInfo(ImmutableMap<String, Field> fields, String idField, Class<T> clazz) {
+    private PersistenceInfo(ImmutableBiMap<String, Field> fields, String idField, Class<T> clazz, String tableName) {
         this.clazz = clazz;
+        this.tableName = tableName;
         this.idField = idField;
         this.column2field = fields;
+        this.field2Column = fields.inverse();
+        this.fieldName2Field = fields.values().stream()
+            .collect(Collectors.toMap(
+                f -> f.getName(),
+                Function.identity()));
     }
 
     public PersistenceInfo<T> aliased(String alias) {
         if (null == alias) {
             return this;
         }
 
-        ImmutableMap.Builder<String, Field> aliasedFields = ImmutableMap.builder();
+        ImmutableBiMap.Builder<String, Field> aliasedFields = ImmutableBiMap.builder();
 
         for (Map.Entry<String, Field> entry : column2field.entrySet()) {
             aliasedFields.put(alias + '.' + removeAlias(entry.getKey()), entry.getValue());
         }
 
-        return new PersistenceInfo<>(aliasedFields.build(), alias + '.' + removeAlias(this.idField), clazz);
+        return new PersistenceInfo<>(aliasedFields.build(), alias + '.' + removeAlias(this.idField), clazz, alias);
     }
 
     private String removeAlias(String key) {
         return key.replaceFirst(".+\\.", "");
     }
+    
+    private static String getTableName(Class<?> clazz) {
+        //TODO: Accomodate table names specified through annotation?
+        
+        return clazz.getSimpleName().toUpperCase();
+    }
+    
+    private static String getIdField(Class<?> clazz) {
+        return Iterators.getOnlyElement(
+            Arrays.stream(clazz.getDeclaredFields())
+                .filter(field -> field.isAnnotationPresent(Id.class))
+                .map(field -> {
+                    Persist p = field.getAnnotation(Persist.class);
```

Comment:
I think stuff like (the `Persist` annotation/parsing) this should stay out of this class. It would be nice if the carbonite annotations, jdo annotations etc were all in their own files and `PersistentInfo` only had the shared logic...